apps/dashboard/src/app/(dashboard)/dashboard/connect/analytics/page.tsx

```diff
@@ -0,0 +1,28 @@
+import { redirect } from "next/navigation";
+import { getApiKeys } from "../../../../api/lib/getAPIKeys";
+import { getAuthToken } from "../../../../api/lib/getAuthToken";
+import { ConnectAnalyticsPage } from "./ConnectAnalyticsPage";
+
+export default async function Page(props: {
+  searchParams: {
+    clientId?: string;
+  };
+}) {
+  const authToken = getAuthToken();
+  const { clientId } = props.searchParams;
+
+  if (!authToken) {
+    const params = new URLSearchParams();
+    params.append("next", "/dashboard/connect/analytics");
+    if (clientId) {
+      params.append("clientId", clientId);
+    }
```

Comment:
does this make it through?

Reply:
oof - yeah this should go inside the `next` search param and not a separate one

Reply:
removing the searchPram thing entirely in favor of proper routing with clientId included in path